tests/Unit/Parallel/Test_Callback.cpp

```diff
@@ -267,6 +285,10 @@ struct RunCallbacks {
     callbacks.emplace_back(
         std::make_unique<Parallel::SimpleActionCallback<
             MultiplyValueByFactor, decltype(proxy_2), double>>(proxy_2, 2.0));
+    const std::optional<bool> callback_2_equal_to_vec_end =
+        callback_2.is_equal_to(*callbacks.back());
+    SPECTRE_PARALLEL_REQUIRE(callback_2_equal_to_vec_end.has_value());
+    SPECTRE_PARALLEL_REQUIRE_FALSE(callback_2_equal_to_vec_end.value());
```

Comment:
[optional] `callback_2_equal_to_vec_end == std::optional{false}` seems clearer to me than these two lines.